lib/semian/atomic_integer.rb

```diff
@@ -0,0 +1,21 @@
+module Semian
+  class AtomicInteger < SharedMemoryObject #:nodoc:
+    attr_accessor :value
+
+    def initialize(name, permissions)
+      @value=0
+    end
+
+    def increase_by(val)
+      @value += val
```

Comment:
hmm feels like this should call `self.value += val`, no? otherwise it's not using the `value=` method that you defined in the module..